src/main/java/com/spotify/futures/CompletableFuturesExtra.java

```diff
@@ -55,4 +60,84 @@ private CompletableFuturesExtra() {
     }
     return new CompletableToListenableFutureWrapper<V>(future);
   }
+
+  public static <T> CompletableFuture<T> immediatelyFailed(Throwable throwable) {
+    final CompletableFuture<T> future = new CompletableFuture<T>();
+    future.completeExceptionally(throwable);
+    return future;
+  }
+
+  public static <T> CompletableFuture<T> exceptionallyCompose(
```

Comment:
About return type: I figured it would be most flexible for the user if it could take CompletionStage as input and return a CompletableFuture.

I realize that this will make future refactorings trickier (we can't go back and weaken the return type), but since this is part of JDK8 I think this is a very small risk. Also, I think CompletionStage and CompletableFuture are basically equivalent anyway - the JDK only has one public implementation.

Regarding naming, I used compose because that's what CompletableFuture uses when your transform function returns a CompletionStage. So it's somewhat consistent with JDK at least.